internal/testutil/treedump.go

```diff
@@ -55,11 +58,32 @@ func TreeDump(dir string) map[string]string {
 		default:
 			return fmt.Errorf("unknown file type %d: %s", ftype, fpath)
 		}
+
+		if ftype != fs.ModeDir {
+			stat, ok := finfo.Sys().(*syscall.Stat_t)
+			if !ok {
+				return fmt.Errorf("cannot get syscall stat info for %q", path)
+			}
+			inode := stat.Ino
+			if len(pathsByInodes[inode]) == 1 {
+				inodes = append(inodes, inode)
+			}
+			pathsByInodes[inode] = append(pathsByInodes[inode], "/"+path)
+		}
 		return nil
 	})
 	if err != nil {
 		panic(err)
 	}
+
+	// Append identifiers to paths who share an inode e.g. hard links.
+	for i := 0; i < len(inodes); i++ {
+		paths := pathsByInodes[inodes[i]]
+		suff := fmt.Sprintf(" <%d>", i+1)
+		for _, path := range paths {
+			result[path] += suff
```

Comment:
Actually I think it is more (or the same) performant and clearer to do `fmt.Printf("%s <%d>", result[path], i+1)`. Normally printf makes it clearer at no expense.